src/opnsense/mvc/app/views/OPNsense/Diagnostics/log.volt

```diff
@@ -99,18 +129,61 @@
               params.push("searchPhrase=" + encodeURIComponent($("input.search-field").val()));
           }
           if ( $('#severity_filter').val().length > 0) {
-              params.push("severity=" + encodeURIComponent($('#severity_filter').val().join(",")));
+              let r_severity = filter_exact ? $('#severity_filter').val() : severities.slice(0,severities.indexOf($('#severity_filter').val()) + 1);
+              params.push("severity=" + encodeURIComponent(r_severity.join(",")));
           }
           if (params.length > 0) {
               download_link = download_link + "?" + params.join("&");
           }
           $('<a></a>').attr('href',download_link).get(0).click();
       });
+
       updateServiceControlUI('{{service}}');
 
       // move filter into action header
       $("#severity_filter_container").detach().prependTo('#grid-log-header > .row > .actionBar > .actions');
+
+
+      function switch_mode(value) {
+          let select = $("#severity_filter");
+
+          // switch select mode and destroy selectpicker
+          select.prop("multiple", filter_exact);
+          select.selectpicker('destroy');
+
+          // remove title option. bug in bs-select. fixed in v1.13.18 https://github.com/snapappointments/bootstrap-select/issues/2491
+          select.find('option.bs-title-option').remove();
+
+          let header_val = filter_exact ? m_header : s_header;
+          select.selectpicker({ header: header_val });
+
+          // attach event handler each time header created
+          $("#exact_severity").on("click", function() {
+              filter_exact = !filter_exact
```

Comment:
```suggestion
          $("#exact_severity").on("click", function(event) {
              event.stopPropagation();
              filter_exact = !filter_exact;
```

`stopPropagation()` allows to to open the selectpicker after recreating it. See also comment below at line 177.

_Note: There was also a semicolon missing in line 162._